app/src/components/annotation/AnnotationSummaryGroup.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useMemo } from "react";
+import { graphql, useFragment } from "react-relay";
+import { css } from "@emotion/react";
+
+import { Flex, Text } from "@phoenix/components";
+import { AnnotationSummaryGroup$key } from "@phoenix/components/annotation/__generated__/AnnotationSummaryGroup.graphql";
+import { AnnotationLabel } from "@phoenix/components/annotation/AnnotationLabel";
+import { AnnotationSummaryPopover } from "@phoenix/components/annotation/AnnotationSummaryPopover";
+import { AnnotationTooltip } from "@phoenix/components/annotation/AnnotationTooltip";
+import {
+  Summary,
+  SummaryValue,
+  SummaryValuePreview,
+} from "@phoenix/pages/project/AnnotationSummary";
+
+const annotationLabelCSS = css`
+  min-height: 20px;
+  align-items: center;
+  justify-content: center;
+  display: flex;
+`;
+
+export const AnnotationSummaryGroup = ({
+  span,
+  showFilterActions = false,
+  renderEmptyState,
+  variant = "pills",
+}: {
+  span: AnnotationSummaryGroup$key;
+  showFilterActions?: boolean;
+  renderEmptyState?: () => React.ReactNode;
+  variant?: "stacked" | "pills";
```

Comment:
Just a gut reaction this feels a bit overloaded - I get that they are the same thing sorta but it's a bit tough since I would never advise to add another variant here and there's other sizing constraints.

Reply:
I've split this up into multiple components with a shared hook